lndr-backend/src/Lndr/Handler/Credit.hs

```diff
@@ -294,3 +297,14 @@ twoPartyBalanceHandler p1 p2 currency = do
     (ServerState pool configTVar _) <- ask
     ucacAddresses <- fmap lndrUcacAddrs . liftIO $ readTVarIO configTVar
     liftIO . withResource pool $ Db.twoPartyBalance p1 p2 (getUcac ucacAddresses currency)
+
+
+multiSettlementHandler :: [CreditRecord] -> LndrHandler NoContent
+multiSettlementHandler transactions = submitMultiSettlement transactions 0
+
+submitMultiSettlement :: [CreditRecord] -> Integer -> LndrHandler NoContent
+submitMultiSettlement [] _ = pure NoContent
```

Comment:
You shouldn't write a general recursive function to do something like this. `fmap (\credit nonceIncrement-> submit...) zipWith credits [0..(length credits)]` would be much shorter and more readable. I need to understand the logic you're trying to implement because this seems a little complicated. Can you explain in the PR comments how MultiSettlement is supposed to work step-by-step? We'll work through a cleaner implementation.